CRM/Report/Form.php

```diff
@@ -4779,6 +4779,16 @@ protected function setOutputMode() {
       FALSE,
       CRM_Utils_Array::value('task', $this->_params)
     ));
+    if ($this->_outputMode !== NULL) {
+      // We didn't get it using output, so we now try with task.
+      $this->_outputMode = str_replace('report_instance.', '', CRM_Utils_Request::retrieve(
+          'task',
+          'String',
+          CRM_Core_DAO::$_nullObject,
+          FALSE,
+          CRM_Utils_Array::value('task', $this->_params)
+        ));
+    }
```

Comment:
The comment and the if condition looks contradictory. Also, I think the `_outputMode` should be set above itself when we get `task` in `$this->_params` if we didn't get it using the `output` request. 

It is set correctly for me due to which I was able to perform the remaining task(print, export, pdf) without having `administer report` permission. Is there something I'm missing ?